src/mono/wasm/runtime/jiterpreter-feature-detect.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    WasmOpcode, WasmSimdOpcode
+} from "./jiterpreter-opcodes";
+import {
+    WasmValtype, WasmBuilder,
+} from "./jiterpreter-support";
+
+export function compileDoJitCall () : WebAssembly.Module {
+    const builder = new WasmBuilder(0);
+    builder.defineType("jit_call_cb", {
+        "cb_data": WasmValtype.i32,
+    }, WasmValtype.void, true);
+    builder.defineType("do_jit_call", {
+        "unused": WasmValtype.i32,
+        "cb_data": WasmValtype.i32,
+        "thrown": WasmValtype.i32,
+    }, WasmValtype.void, true);
+    builder.defineImportedFunction(
+        "i", "jit_call_cb", "jit_call_cb", true,
+        undefined
+    );
+    builder.defineFunction({
+        type: "do_jit_call",
+        name: "do_jit_call_indirect",
+        export: true,
+        locals: {}
+    }, () => {
+        builder.block(WasmValtype.void, WasmOpcode.try_);
+        builder.local("cb_data");
+        builder.callImport("jit_call_cb");
+        builder.appendU8(WasmOpcode.catch_all);
+        builder.local("thrown");
+        builder.i32_const(1);
+        builder.appendU8(WasmOpcode.i32_store);
+        builder.appendMemarg(0, 0);
+        builder.endBlock();
+        builder.appendU8(WasmOpcode.end);
+    });
+    // Magic number and version
+    builder.appendU32(0x6d736100);
+    builder.appendU32(1);
+    builder.generateTypeSection();
+    builder.emitImportsAndFunctions(false);
+    const buffer = builder.getArrayView();
+    // console.log(`do_jit_call generated: ${JSON.stringify(buffer)}`);
+    return new WebAssembly.Module(buffer);
+}
+
+export function compileSimdFeatureDetect () : WebAssembly.Module {
+    const builder = new WasmBuilder(0);
+    builder.defineType("test", {}, WasmValtype.void, true);
+    builder.defineFunction({
+        type: "test",
+        name: "test",
+        export: true,
+        locals: {}
+    }, () => {
+        builder.i32_const(0);
+        builder.appendSimd(WasmSimdOpcode.i32x4_splat);
+        builder.appendU8(WasmOpcode.drop);
+        builder.appendU8(WasmOpcode.end);
+    });
+    // Magic number and version
+    builder.appendU32(0x6d736100);
```

Comment:
do method names like `appendU32` minify well with rollup ?

Reply:
I'm not sure what you mean

Reply:
how many bytes we are adding to size of JS with this PR ?

Reply:
each feature detect generator is around 400 bytes of uncompressed js after minification, and i'm pretty sure brotli will crunch it up just fine. we had a decent amount of code before this that it's replacing, so i think it's ok

Reply:
https://github.com/dotnet/runtime/issues/86447